Integrations/src/test/java/io/deephaven/integrations/learn/ComputerTest.java

```diff
@@ -0,0 +1,126 @@
+        return new Input[]{new Input(new String[]{"Column1","Column2"}, gatherFunc), new Input("Column3", gatherFunc)};
+    }
+
+    @Test
+    public void computeMethodTest() {
+
+        final Function<Object[], Object> func = args -> args;
+        final int batchSize = 7;
+
+        Computer computer = new Computer(table, func, inputs, batchSize);
+
+        for (int i = 0 ; i < 9 ; i++) {
+            computer.compute(i);
+        }
+
+        computer.clear();
+    }
+
+    @Test(expected = io.deephaven.base.verify.RequirementFailure.class)
+    public void nullTableTest() {
+
+        final Function<Object[], Object> func = args -> args;
+        final int batchSize = 7;
+
+        Computer computer = new Computer(null, func, inputs, batchSize);
+    }
+
+    @Test(expected = io.deephaven.base.verify.RequirementFailure.class)
+    public void nullFunctionTest() {
+
+        final Function<Object[], Object> func = null;
+        final int batchSize = 7;
+
+        Computer computer = new Computer(table, func, inputs, batchSize);
+    }
+
+    @Test(expected = io.deephaven.base.verify.RequirementFailure.class)
+    public void invalidBatchSizeTest() {
+
+        final Function<Object[], Object> func = args -> args;
+        final int batchSize = 0;
+
+        Computer computer = new Computer(table, func, inputs, batchSize);
+    }
+
+    @Test
+    public void computeGatherGetTest() {
+
+        final int batchSize = 7;
+
+        final IndexSet[] indexSetTarget = new IndexSet[1];
+        ColumnSource<?>[][] colSourceTarget = new ColumnSource[inputs.length][];
+        for (int i = 0 ; i < inputs.length ; i++) {
+            colSourceTarget[i] = inputs[i].createColumnSource(table);
+        }
+
+        final int[] counter = {0};
+
+        Function<Object[], Object> myGather = (params) ->
+        {
+            // first we test equality on IndexSet
+            Assert.assertEquals(indexSetTarget[0], params[0]);
+
+            // now we test equality on ColumnSources, which depends on which input object we are comparing
+            if (counter[0] == 0) {
+
+                Assert.assertTrue(Objects.deepEquals(new Object[]{colSourceTarget[0]}[0], params[1]));
+                counter[0]++;
+
+            } else {
+                Assert.assertTrue(Objects.deepEquals(new Object[]{colSourceTarget[1]}[0], params[1]));
+            }
+
+            return 4;
+        };
+
+        Function<Object[], Object> myModel = (thisInput) ->
+        {
+            Assert.assertArrayEquals(new Object[]{4,4}, thisInput); // I have 2 Input objects in inputs, so gather gets called twice
+            return 5;
+        };
+
+        Input[] thisInput = createInputs(myGather);
+
+        Computer computer = new Computer(table, myModel, thisInput, batchSize);
+
+        for (int i = 0 ; i < 9 ; i++) {
+            computer.compute(i);
+        }
+        for (int i = 0 ; i < 9 ; i++) {
+            indexSetTarget[0] = computer.getFuture().getIndexSet();
```

Comment:
I would add a comment that calling `computer.getFuture().get()` triggers assertions in the gather and model functions.